lib/roo_on_rails/context_logging.rb

```diff
@@ -0,0 +1,80 @@
+require 'active_support/core_ext/module/delegation'
+require 'active_support/core_ext/object/blank'
+require 'logger'
+require 'active_support/logger'
+
+module RooOnRails
+  # Wraps any standard logger to provide context, similar to `ActiveSupport::TaggedLogging`
+  # but with key/value pairs that are appended to the end of the text.
+  #
+  #   logger = RooOnRails::ContextLogging.new(Logger.new($stdout))
+  #   logger.context(a: 1, b: 2) { logger.info 'foo' }                     # Logs "foo -- a=1 b=2"
+  #   logger.context(a: 1) { logger.context(b: 2) { logger.info('foo') } } # Logs "foo -- a=1 b=2"
+  #
+  # You can also use `push_context` and `pop_context` in before/after blocks.
+  #
+  #   class ApplicationController < ActionController::Base
+  #     before_action { logger.push_context(user_id: current_user.id) }
+  #     after_action { logger.pop_context }
+  #   end
+  module ContextLogging
+    module Formatter
+      def call(severity, timestamp, progname, msg)
```

Comment:
Other libraries (e.g. [Slog](https://github.com/sczizzo/slog)) or platforms (e.g. [this](https://github.com/sirupsen/logrus#example)) tend to follow this format instead:

```
$logger.with(key: 'value').info("human readable message")
```